bundle/src/main/java/com/adobe/acs/commons/packaging/util/AssetPackageUtil.java

```diff
@@ -0,0 +1,225 @@
+        this.assetExclusionPatterns = generatePatterns(properties.get(PN_ASSET_EXCLUSIONS, new String[]{}));
+        this.excludedPages = new LinkedHashSet<>();
+    }
+
+    /**
+     * Generates the filter paths by recursing through the content and returns the list.
+     *
+     * @return the compiled list of {@link PathFilterSet} objects with assets and pages.
+     */
+    public List<PathFilterSet> getPackageFilterPaths() {
+        final String pagePath = properties.get(PN_PAGE_PATH, String.class);
+        final Set<PathFilterSet> packageFilterPathSet;
+        if (StringUtils.isBlank(pagePath)) {
+            packageFilterPathSet = Collections.emptySet();
+        } else {
+            packageFilterPathSet = this.findAssetPaths(resourceResolver, pagePath);
+        }
+
+        // Add the page path as a filter unless it is explicitly excluded
+        this.addPagePath(packageFilterPathSet, pagePath, properties.get(PN_EXCLUDE_PAGES, false));
+
+        List<PathFilterSet> packageFilterPaths = new ArrayList<>(packageFilterPathSet);
+        // Reverse the new list to put the page inclusion at the top
+        Collections.reverse(packageFilterPaths);
+        return packageFilterPaths;
+    }
+
+    /**
+     * Takes a list of Strings that should be valid patterns. If they are not then the exception is
+     * logged and the pattern is not added to the list of checked patterns.
+     *
+     * @param input The array of intended patterns
+     * @return The converted array of patterns
+     */
+    private List<Pattern> generatePatterns(String[] input) {
+        final List<Pattern> patterns = new ArrayList<>();
+        for (String item : input) {
+            try {
+                patterns.add(Pattern.compile(item));
+            } catch (PatternSyntaxException e) {
+                log.error("Pattern invalid, skipping. Pattern value: " + item);
+            }
+        }
+        return patterns;
+    }
+
+    /**
+     * Recursively iterate over the parent path specified in the configuration to aggregate all the
+     * String or String Array property values that are referencing a path in the DAM.
+     *
+     * @param resourceResolver Resolver from the current Request
+     * @param pagePath         The page path
+     * @return The full list of filter sets
+     */
+    private Set<PathFilterSet> findAssetPaths(final ResourceResolver resourceResolver,
+                                               final String pagePath) {
+
+        final Set<PathFilterSet> filters = new LinkedHashSet<>();
+        final Resource parentResource = resourceResolver.resolve(pagePath);
+        if (isExcluded(this.pageExclusionPatterns, pagePath)) {
+            final PageManager pageManager = resourceResolver.adaptTo(PageManager.class);
+            Page page = pageManager.getContainingPage(parentResource);
+            excludedPages.add(page.getPath());
```

Comment:
One last item in testing.

The **page** exclude patterns are AEM **paths**, but the **asset** exclude patterns are **regexp** expressions.  This is confusing from a usability perspective because the fields appear to be 100% the same (just one functions on pages, the other on assets).  I had to figure this out via trial and error when testing.

Perhaps we can solve with additional instructions and differentiated naming.  Or maybe you have a better idea.

![Screen Shot 2019-05-14 at 3 40 16 PM](https://user-images.githubusercontent.com/972809/57730912-1d79c100-765f-11e9-8c60-a225ca6a4dcd.png)

Reply:
The page exclusion patterns accept regex as well. They are just represented in the filters as the matched page paths. Are you seeing the page exclusion regex not working? It's all run through `isExcluded` in `AssetPackageUtil` - line 217

Reply:
Addressed in the latest commit